sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py

```diff
@@ -158,11 +158,15 @@ def _from_cncf_events(event):
         raise_with_traceback(ValueError, msg, err)
 
 
-def _build_request(endpoint, content_type, events):
+def _build_request(endpoint, content_type, events, **kwargs):
+    channel_name = kwargs.pop('channel_name', None)
     serialize = Serializer()
     header_parameters = {}  # type: Dict[str, Any]
     header_parameters['Content-Type'] = serialize.header("content_type", content_type, 'str')
 
+    if channel_name:
+        header_parameters['aeg-channel-name'] = serialize.header("channel_name", channel_name, 'str')
```

Comment:
What rules are there for escaping the channel name? I'm not sure why we need to do this in py3 in the first place for strings - wasn't this just for unicode/utf-8 handling?

Reply:
we wouldn't need it in python3 - this can go full py3 mode now that we are using keyword only instead of kwarg